src/app/features/tech-record/components/tech-record-summary-changes/tech-record-summary-changes.component.ts

```diff
@@ -205,8 +205,8 @@ export class TechRecordSummaryChangesComponent implements OnInit, OnDestroy {
   }
 
   isNotEmpty(value: unknown): boolean {
-    if (value == null || value === '') return false;
-    if (typeof value === 'object') return Object.values(value).length > 0;
+    if (value === '') return false;
+    if (typeof value === 'object' && value !== null) return Object.values(value).length > 0;
```

Comment:
had to check that value is not null here otherwise it won't drop through to the final return

Reply:
Just want to double check if this is an issue or not, this checks for exact equality to null, and thus will allow undefined - is that the behaviour you want?

Reply:
@pbardy2000 So we will only be wanting to let null through as the resolver is nulling the fields that contain bad data rather than setting them to undefined. I'll extend the empty string check to check for undefined

Reply:
Is there a test for this that has an explicit null value ?

Reply:
Ive got 2 things here:

1. lets add a test for an object with null to be allowed througj

2. We're now saying that we allow null but not empty in a function called isNotEmpty which would raise some questions in the future if I saw this again. In the object world this would break the O in solid, even more so that it was discussed that this would move into a more utils type file in the future.

Hence is there a more appropriate name and can we self document the code, or do we default to a comment ?

Reply:
Both fair points Tom I'll work on those